### ISSUE TITLE
Feature Request: Easy access to neighboring node not on the boundary for LR-meshes

### ISSUE THREAD

**Issue by kmokstad** [OPM/IFEM#203]
To be able to run adaptive C1-continuous plate problems (KL) with symmetry boundary conditions (the problem I showed on the whiteboard today), I need to constrain the node that is adjacent to a boundary node, but not on the boundary itself (dvs. noden innafor). That is, we need to do something like ASMs2DC1::constrainEdge but for ASMu2D. Maybe we need a ASMu2DC1 class for that.

**Comment by akva2** [comment]
The function used to obtain boundary nodes in LR has a depth param, maybe this is enough? Ie depth = 1

**Comment by akva2** [comment]
yes, essentially do what is done in ASMu2D::getEdgeNodes, but set depth = 2. after sort you have the nodes in interleaved fashion. but which is first (the node on the boundary, or the one inside) will depend on the edge in question, ie for left boundary you get the node on the boundary, then the one inside, on the right boundary it is the opposite.

**Comment by akva2** [comment]
```
this->getBasis(basis)->getEdgeFunctions(edgeFunctions2,
                                          static_cast<LR::parameterEdge>(edge),2);
 ASMunstruct::Sort(u, v, orient, edgeFunctions2);
 for (auto& it : edgeFunctions2)
    std::cout << it->getId() << " " << this->getCoord(it->getId()+1) << std::endl;
```

**Comment by kmokstad** [comment]
OK, thanks. I will give it a try.

**Comment by VikingScientist** [comment]
![c1-inner-functions](https://user-images.githubusercontent.com/3500376/31333575-e20e5f3e-acea-11e7-9798-1e108e9f10c6.png)

**Comment by VikingScientist** [comment]
This is not trivial to do in the general case. However, with the proposed solution in #207 you are guaranteed some regular structure on the boundary interior (well that's a contradiction if I ever saw one, but I think you know what I mean) which might make it easier to pick corresponding "pairs" of functions that form normal derivatives.
